routes/index.js

```diff
@@ -118,22 +117,72 @@ function formatUserData (res) {
       flist[i] = fname + ' ' + lname
     }
   }
+  userData.id = friends[0].id
   userData.firstname = friends[0].firstname.charAt(0).toUpperCase() + friends[0].firstname.slice(1)
   userData.lastname = friends[0].lastname.charAt(0).toUpperCase() + friends[0].lastname.slice(1)
   userData.friendslist = flist
   return userData
 }
 
 // Route for adding friends
-router.post('/api/friendships', function (req, res) {
-  // const { firstname, lastname, email } = req.body
-
-  // check if duplicate
-
-  // add user to users table
-
-  // add friendship to friendship table (friender id, id of added friend)
+router.post('/api/friendships', async function (req, res) {
```

Comment:
The reason this route is so messy is because it's hella indented and hard to read at a glance because
* you're duplicating code across your first if-statement
* you're nesting promises, rather than using async/await.

Your code has two steps that are separate, so keep them separate, not nested. IE
```
// STEP 1 - get userId
let userId
const userExists = await emailAlreadyRegistered(friendemail)
if (userExists) 
  userId = await create user
else 
  userId = await get user id

// STEP 2 - add friend
if (userExists) 
  const friendshipExists = await checkFriendshipExists
  if (friendshipExists):
    return 409

add friend, return 201 - created

```